applications/vanilla/settings/class.hooks.php

```diff
@@ -583,7 +583,11 @@ public function userModel_getCategoryViewPermission_create($sender) {
             }
 
             $options = ['ForeignID' => $permissionCategoryID];
-            $result = Gdn::userModel()->checkPermission($userID, $permission, $options);
+
+            $result = (Gdn::session()->isValid())
+                ? Gdn::session()->checkPermission($permission)
```

Comment:
Some addons (e.g. Category Roles) use the "loadPermissions" event to augment a user's permissions. This event is fired in `UserModel::getPermissions`, which is called by `UserModel::checkPermission`. I don't believe that event would be fired from `Gdn_Session`. Could Ranks be updated to use the "loadPermissions" event, at least for category permissions?

Reply:
Using the "loadPermissions" event i'm able to merge the sessions permissions with the users permission in a similar way that Category Roles did.  Only possible issue I could see is if caching isn't enabled, we default back to the permissions set in the db.  Do you think this could be an issue?
```   
 } elseif (!Gdn::cache()->activeEnabled() && is_array($user->Permissions)) {
            // Only attempt to use the DB field value if permissions aren't being cached elsewhere.
            $permissions = new Vanilla\Permissions($user->Permissions);
```
https://github.com/vanilla/vanilla/blob/master/applications/dashboard/models/class.usermodel.php#L286